src/smart-components/common/page-action-route.js

```diff
@@ -0,0 +1,16 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Route } from 'react-router-dom';
+
+const PageActionRoute = ({ pageAction, ...props }) => {
+  useEffect(() => {
+    insights.chrome.appAction(pageAction);
```

Comment:
You should clean once the route unmounts as well `return () => insights.chrome.appAction(undefined);`

Reply:
already there!